### PR TITLE
chore: restore congrats page links

### PR DESCRIPTION
**Pull request by EugeneDevv**

### CODE REVIEW

**Review by Paulmburu — APPROVED** [review]

File: lib/presentation/onboarding/set_nickname/pages/congratulations_page.dart [hunk 1]
```diff
@@ -31,122 +38,290 @@ class _SetNickNamePageState extends State<SetNickNamePage> {
...
-                          if (value!.isEmpty) {
-                            return nameInputValidateString;
-                          }
-                        },
-                        onChanged: (String val) {
-                          setState(() {
-                            nickName = val;
-                          });
+                      smallVerticalSizedBox,
+                      StoreConnector<AppState, AppStateViewModel>(
```
Comment (KellyEgesa) [inline lib/presentation/onboarding/set_nickname/pages/congratulations_page.dart:111]: Remove this, It is repetition

**Review by KellyEgesa — CHANGES_REQUESTED** [review]

**Comment by EugeneDevv** [comment]
shot1
shot2

**Review by KellyEgesa — APPROVED** [review]

File: lib/presentation/content/pages/content_details_page.dart [hunk 2]
```diff
@@ -26,17 +26,19 @@ class ContentDetailPage extends StatelessWidget {
...
+    required this.payload,
+  });
 
-  final Content contentDetails;
+  final ContentDetails payload;
 
   @override
   Widget build(BuildContext context) {
-    final Widget publishDate = contentDetails.metadata!.createdAt!.isNotEmpty
+    final Widget publishDate = payload.content.metadata!.createdAt!.isNotEmpty
```
Comment (johnngugi) [inline lib/presentation/content/pages/content_details_page.dart:37]: Potential for app crashing due to the null assertion operators

File: lib/presentation/content/pages/content_details_page.dart [hunk 3]
```diff
@@ -46,17 +48,17 @@ class ContentDetailPage extends StatelessWidget {
...
                   width: MediaQuery.of(context).size.width,
                   child: ChewieVideoPlayer(
                     autoPlay: true,
                     chewieController: initializeChewiController(
                       dataSource:
-                          contentDetails.featuredMedia?.first?.mediaUrl ?? '',
+                          payload.content.featuredMedia?.first?.mediaUrl ?? '',
                     ),
-                    thumbnail: contentDetails.heroImage!.url!,
+                    thumbnail: payload.content.heroImage!.url!,
```
Comment (johnngugi) [inline lib/presentation/content/pages/content_details_page.dart:61]: Avoid null assertion operators

File: lib/presentation/content/pages/content_details_page.dart [hunk 4]
```diff
@@ -69,7 +71,7 @@ class ContentDetailPage extends StatelessWidget {
                       // TODO(abiud): replace with cached network image to
                       // handle showing an image before the network one loads
                       image: NetworkImage(
-                        contentDetails.heroImage!.url!,
+                        payload.content.heroImage!.url!,
```
Comment (johnngugi) [inline lib/presentation/content/pages/content_details_page.dart:74]: null assertion operators

File: lib/presentation/content/pages/content_details_page.dart [hunk 5]
```diff
@@ -155,39 +157,40 @@ class ContentDetailPage extends StatelessWidget {
                       ],
                     ),
                     EstimatedReadTimeBadge(
-                      estimateReadTime: contentDetails.estimate!,
-                      contentType: contentDetails.contentType!,
+                      estimateReadTime: payload.content.estimate!,
```
Comment (johnngugi) [inline lib/presentation/content/pages/content_details_page.dart:160]: Same as before

File: lib/presentation/content/pages/content_details_page.dart [hunk 6]
```diff
@@ -155,39 +157,40 @@ class ContentDetailPage extends StatelessWidget {
...
-                  ),
-                )
+                if (payload.showReactions)
+                  Center(
+                    child: Wrap(
+                      alignment: WrapAlignment.center,
+                      runSpacing: 15,
+                      children: <Widget>[
+                        LikeContentWidget(
+                          contentID: payload.content.contentID!,
```
Comment (johnngugi) [inline lib/presentation/content/pages/content_details_page.dart:173]: Same as before

File: lib/presentation/content/widgets/mini_content_widget.dart [hunk 7]
```diff
@@ -1,35 +1,47 @@
...
-        padding: const EdgeInsets.all(8.0),
-        child: Align(
-          alignment: Alignment.centerLeft,
-          child: Text(
-            importantLinkString,
-            style: TextThemes.boldSize16Text(Theme.of(context).primaryColor),
+    if (contentDetails?.title?.isNotEmpty ?? false) {
+      return GestureDetector(
+        onTap: () => Navigator.of(context)
+            .pushNamed(BWRoutes.contentDetailPage, arguments: ContentDetails(content: contentDetails!, showReactions: false)),
```
Comment (johnngugi) [inline lib/presentation/content/widgets/mini_content_widget.dart:20]: Same as before

File: lib/presentation/onboarding/set_nickname/pages/congratulations_page.dart [hunk 8]
```diff
@@ -31,122 +38,276 @@ class _SetNickNamePageState extends State<SetNickNamePage> {
...
+          ),
+          floatingActionButton: Container(
+            padding: const EdgeInsets.symmetric(horizontal: 30),
+            width: double.infinity,
+            height: 52,
+            child: MyAfyaHubPrimaryButton(
+              buttonKey: continueKey,
+              onPressed: vm.appState.wait!.isWaitingFor(setNickNameFlag)
+                  ? null
+                  : () async {
```
Comment (johnngugi) [inline lib/presentation/onboarding/set_nickname/pages/congratulations_page.dart:278]: Extract this into a named function

File: lib/presentation/onboarding/set_nickname/pages/congratulations_page.dart [hunk 9]
```diff
@@ -31,122 +38,276 @@ class _SetNickNamePageState extends State<SetNickNamePage> {
...
+                      final bool? isFormValid =
+                          _formKey.currentState?.validate();
+                      if (!InternetConnectivitySubject()
```
Comment (johnngugi) [inline lib/presentation/onboarding/set_nickname/pages/congratulations_page.dart:281]: `InternetConnectivitySubject` is no longer used. Refer to [this](https://github.com/savannahghi/myafyahub/blob/5adb026206086f4afb9716da1e8ea8ce266193b9/lib/presentation/onboarding/login/pages/login_page.dart#L190) to see how connectivity is checked

**Review by johnngugi — CHANGES_REQUESTED** [review]

**Review by johnngugi — APPROVED** [review]

**Review by KellyEgesa — APPROVED** [review]

**Pull request merged**